libsql/src/hrana/hyper.rs

```diff
@@ -142,8 +142,8 @@ impl Conn for HttpConnection<HttpSender> {
         })
     }
 
-    async fn is_autocommit(&self) -> crate::Result<bool> {
-        Ok(true) // connection without transaction always commits at the end of execution step
+    fn is_autocommit(&self) -> bool {
+        true // connection without transaction always commits at the end of execution step
```

Comment:
this is not true, one can always execute `BEGIN TRANSACTION` on a connection, then this would be a bug

Reply:
Isn't server-side transaction mapped 1-1 to client-side stream? Atm. if you do `conn.execute("BEGIN TRANSACTION")` it will open a new stream, then immediately send a pair of `"execute: {my_sql_stmt}", "close"` Hrana HTTP requests: I assumed that the close request immediately calls for any resources related to the stream (including transaction) to be released.